bindings/python/pyiree/rt/system_api.py

```diff
@@ -125,6 +133,41 @@ def _get_global_config():
   return _global_config
 
 
+def _normalize_numpy(array: np.ndarray,
+                     single_precision: bool = False) -> np.ndarray:
+  if np.isscalar(array):
+    array = np.array(array)
+
+  # IREE models booleans as int8s.
+  if array.dtype == np.bool:
+    array = array.astype(np.int8)
+
+  if single_precision:
+    if array.dtype == np.float16 or array.dtype == np.float64:
+      array = array.astype(np.float32)
+    if array.dtype == np.int16 or array.dtype == np.int64:
+      array = array.astype(np.int32)
+  return array
+
+
+def normalize_value(value: Any) -> Optional[np.ndarray]:
+  """Normalizes the given value for input to (or comparison with) IREE."""
+  if isinstance(value, np.ndarray):
```

Comment:
I feel like there is an important, more duck-typey idiom for this, but I'm having trouble placing it now: there are a lot of array-like things that can exist.

Reply:
Simplified things a bit. I vaguely remember hearing about that, not sure if it's [this](https://numpy.org/neps/nep-0030-duck-array-protocol.html)? In any case this should do for now, and using `np.asarray` instead of `.numpy()` lets avoid some of the poor choices tensorflow makes there.